### PR TITLE
grpc patch: Fixed NAN/INFINITY definitions to work on Windows 11 SDK.

### PR DESCRIPTION
**Pull request by Altina-oz**

### CODE REVIEW

**Comment by Altina-oz** [comment]
in MacOS CI:
error: @programdir/core/sandbox/modules/os.lua:273: ld: warning: search path '/usr/local/Cellar/gcc/14.2.0/lib/gcc/14/' not found

**Comment by star-hengxing** [comment]
To support cross-compilation, we need to patch cmake and build host protobuf (get protoc) & target protobuf.

ref: https://github.com/xmake-io/xmake-repo/pull/5831

```lua
-- target
add_deps("protobuf-cpp")
-- host protobuf
if package:is_cross() then
    package:add("deps", "protoc")
end
```

File: packages/g/grpc/xmake.lua [hunk 1]
```diff
@@ -54,6 +54,9 @@ package("grpc")
             "-DgRPC_PROTOBUF_PACKAGE_TYPE=CONFIG"}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "Release"))
         table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        if package:is_cross() then
+            package:add("deps", "protoc")
```
[the remark this answers is not in the document]
Comment (Altina-oz) [reply packages/g/grpc/xmake.lua]: 嗯刚才发现加错了hh

**Comment by star-hengxing** [comment]
We need to make a patch, see https://github.com/grpc/grpc/issues/36654

**Pull request merged**